src/app.rs

```diff
@@ -163,10 +163,18 @@ impl<'a, 'b> Application<'a, 'b> {
                 )
                 {
                     let mut input = engine.world.write_resource::<InputHandler>();
-                    if let Event::WindowEvent { ref event, .. } = event {
-                        input.send_event(&event);
+                    if let WinitEvent::WindowEvent { ref event, .. } = event {
+                        input.send_event(event);
                     }
                 }
+                let event = match event {
```

Comment:
I'd actually make the hierarchy even flatter, but it's definitely an improvement.